lib/apiservers/engine/network/utils.go

```diff
@@ -52,24 +52,29 @@ var (
 
 	cbpLock         sync.Mutex
 	ContainerByPort map[string]string // port:containerID
+	once            sync.Once
 )
 
 func init() {
 	portMapper = portmap.NewPortMapper()
 	btbRules = make(map[string][]string)
 	ContainerByPort = make(map[string]string)
+}
 
-	l, err := netlink.LinkByName(publicIfaceName)
-	if l == nil {
-		l, err = netlink.LinkByAlias(publicIfaceName)
-		if err != nil {
-			log.Errorf("interface %s not found", publicIfaceName)
-			return
+func Init() {
+	once.Do(func() {
```

Comment:
To reduce the risk of someone forgetting to call `Init`, would it make sense to call it at the beginning of each of the public methods in the file? (Or is that just overkill?)

Reply:
I think it's overkill. An alternative would be to keep `init` but change where the error message comes out to when we use `publicIFaceName` but that _pattern_ means we're still doing undesired initialization when consumed as an in-tree provider.

I think the only neat way of doing this is to have an interface, an unexported type that implements the interface, and a `New` function that constructs the type while ensuring the constructor code is called. However that could be a Go anti-pattern for all I know.